src/main/java/wolox/training/repositories/BookRepository.java

```diff
@@ -1,8 +1,20 @@
 package wolox.training.repositories;
 
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.Repository;
 import wolox.training.models.Book;
 
 public interface BookRepository extends Repository<Book, Long> {
-    Book findFirstByAuthor (String author);
+    Optional<Book> findFirstByAuthor (String author);
```

Comment:
use final here

Reply:
in the param?

Reply:
exactly, below I left you a link to see the advantges of using final.